packages/material/scss/button/_layout.scss

```diff
@@ -1,36 +1,41 @@
-$bare-button-hover-opacity: .12 !default;
-$outline-button-border-width: 2px;
+$flat-button-hover-opacity: .12 !default;
+$flat-button-focused-opacity: .12 !default;
+$flat-button-active-opacity: .16 !default;
+$flat-button-selected-opacity: .16 !default;
+$outline-button-border-width: 2px !default;
+
 @import "~@progress/kendo-theme-default/scss/button/layout";
 
 @include exports("button/layout/material") {
-    .k-button.k-outline {
+
+    .k-button {
         &::before {
-            content: "";
             display: block;
-            opacity: 0;
-            pointer-events: none;
-            position: absolute;
-            z-index: 0;
-            top: -$button-border-width;
-            left: -$button-border-width;
-            right: -$button-border-width;
-            bottom: -$button-border-width;
+            transition: opacity .4s cubic-bezier(.25, .8, .25, 1);
         }
     }
 
-    .k-button-group {
-        &.k-button-group-outline {
-            .k-button {
-                border-width: $outline-button-border-width;
-                border-style: solid;
-                box-sizing: border-box;
-                padding-top: $button-padding-y / 2;
-                padding-bottom: $button-padding-y / 2;
-            }
 
-            .k-button ~ .k-button {
-                margin-left: -$outline-button-border-width;
-            }
+    // Outline button
+    .k-button.k-outline {
+        padding-top: calc( #{$button-padding-y} - #{$outline-button-border-width} );
+        padding-bottom: calc( #{$button-padding-y} - #{$outline-button-border-width} );
+        padding-left: calc( #{$button-padding-x} - #{$outline-button-border-width} );
+        padding-right: calc( #{$button-padding-x} - #{$outline-button-border-width} );
+        border: $outline-button-border-width solid currentColor;
+    }
+
+
+    // Outline button group
+    .k-button-group-outline {
+        .k-button ~ .k-button {
+            border-left-width: 0;
```

Comment:
I think the approach that used `margin-left` / `-right` has a benefit that we can raise the `z-index` of the hovered / active button. This way changing the `currentColor` of these states will show the full border of the button. Does this make sense?

Reply:
It was like that.

However, in disabled groups, Material guidelines require translucent color. Having two opacity stacked on top each other gives a nasty visual artifacts.

I am OK with reverting the behaviour. It yields cleaner code.